plugins/resin/src/main/java/com/navercorp/pinpoint/plugin/resin/interceptor/HttpRequestInterceptor.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2014 NAVER Corp.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.navercorp.pinpoint.plugin.resin.interceptor;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.navercorp.pinpoint.bootstrap.context.Trace;
+import com.navercorp.pinpoint.bootstrap.instrument.InstrumentMethod;
+import com.navercorp.pinpoint.bootstrap.interceptor.AroundInterceptor;
+import com.navercorp.pinpoint.bootstrap.logging.PLogger;
+import com.navercorp.pinpoint.bootstrap.logging.PLoggerFactory;
+import com.navercorp.pinpoint.plugin.resin.AsyncAccessor;
+import com.navercorp.pinpoint.plugin.resin.TraceAccessor;
+
+/**
+ * 
+ * @author huangpengjie@fang.com
+ *
+ */
+public class HttpRequestInterceptor implements AroundInterceptor {
+
+	private PLogger logger = PLoggerFactory.getLogger(this.getClass());
+
+	private InstrumentMethod targetMethod;
+
+	public HttpRequestInterceptor(InstrumentMethod targetMethod) {
+		this.targetMethod = targetMethod;
+	}
+
+	@Override
+	public void before(Object target, Object[] args) {
+		logger.beforeInterceptor(target, target.getClass().getName(), targetMethod.getName(), "", args);
+		try {
+			com.caucho.server.http.HttpRequest httpRequest = (com.caucho.server.http.HttpRequest) target;
+			HttpServletRequest httpServletRequest = (HttpServletRequest)httpRequest.getRequestFacade();
```

Comment:
`com.caucho.server.http.HttpRequest` (and it's parent class `AbstractHttpRequest`) in Resin 3.x does not seem to have `getRequestFacade()` method and would throw exceptions when running the agent on Resin 3.x.

You could attach this interceptor to hook on `com.caucho.server.http.HttpServletRequestImpl.finishRequest()`. The `target` object would then be an instance of `AsyncAccessor` and `TraceAccessor` and you'll be able to use the methods directly like so
```
if(target != null){
    if (target instanceof AsyncAccessor) {
        // reset
        ((AsyncAccessor) target)._$PINPOINT$_setAsync(Boolean.FALSE);
    }
    if (target instanceof TraceAccessor) {
        ...
```